hyperspy/roi.py

```diff
@@ -607,6 +607,56 @@ def _top_changed(self, old, new):
         else:
             self.update()
 
+    @property
+    def width(self):
+        return self.right - self.left
+
+    @width.setter
+    def width(self, value):
+        if value == self.width:
+            return
+        self.right -= self.width - value
+
+    @property
+    def height(self):
+        return self.bottom - self.top
+
+    @height.setter
+    def height(self, value):
+        if value == self.height:
+            return
+        self.bottom -= self.height - value
+
+    @property
+    def x(self):
+        return self.left
+
+    @x.setter
+    def x(self, value):
+        if value != self.x:
+            diff = value - self.x
+            self._applying_widget_change = True
+            with self.events.changed.suppress():
+                self.right += diff
+                self.left += diff
```

Comment:
Here you should set `self._bounds_check` to `False` in a `try`-`finally` block so that it also handles negative diff (`_applying_widget_change` should also benefit from that block).